app/presenters/keyword_presenter.rb

```diff
@@ -12,6 +12,18 @@ def matching_adword_urls
     return unless defined?(filter_params[:adwords_url_contains]) && filter_params[:adwords_url_contains]
 
     keyword.ads_top_urls.select { |item| item.downcase.include?(filter_params[:adwords_url_contains].downcase) }
+  rescue NoMethodError
+    nil
+  end
+
+  def matching_result_urls
+    return unless filter_params[:word] && filter_params[:match_at_least]
+
+    keyword.result_urls.select { |item|
+      item.downcase.match("\\w*(#{filter_params[:word]}.*){#{filter_params[:match_at_least]},}")
```

Comment:
Are the filters already down-cased?
if not, we must do that outside from the select loop 😇